partcad/src/partcad/shape.py

```diff
@@ -72,15 +72,17 @@ def _finalize_real(self, show_object):
         if not show_object is None:
             self.show(show_object)
 
-    def render_svg_somewhere(self, project=None, filepath=None):
+    def render_svg_somewhere(self,  project=None, filepath=None, svg_render_opts=None):
         """Renders an SVG file somewhere and ignore the project settings"""
         if filepath is None:
             filepath = tempfile.mktemp(".svg")
 
+        if svg_render_opts is None:
```

Comment:
I would suggest to use 'project.ctx' here to extract the options and to apply them, instead of passing the opts structure. The parameter 'project' is provided if and only if the project settings need to respected.

Reply:
yes this will results as you expected. If we have configs in yaml file, then it will be reflected. otherwise default will be applied.